### PR TITLE
feat(task): refactor task interface, introduce task runtime component

### PR DESCRIPTION
**Pull request by LioRoger**
#### What type of PR is this?

type-feature
#### What this PR does / why we need it:
1. remove BaseTask, introduce TaskBase only contains jobContext variables
2. introduce TaskContainer to hold task runtime info. TaskStatus will not known by task anymore.
3. move cloudStorageService to TaskContext as a candidate component
4. move TaskMonitor/DefaultTaskBuilder eg to agent package
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by LuckyPickleZZ — APPROVED** [review]

File: server/odc-server/pom.xml [hunk 1]
```diff
@@ -148,6 +148,11 @@
             <artifactId>mockito-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
```
Comment (yhilmare) [inline server/odc-server/pom.xml:152]: why should we need this?

**Review by guowl3 — DISMISSED** [review]
lgtm

File: server/odc-server/pom.xml (hunk 1 above)
Comment (LioRoger) [reply server/odc-server/pom.xml:152]: mock static and MockVerify need this package

**Review by yhilmare — DISMISSED** [review]
lgtm

**Review by yizhouxw — DISMISSED** [review]
lgtm

**Review by MarkPotato777 — COMMENTED** [review]
The refactor looks good to me. I have another requirement but it's not in emergency: Can we have a mechanism to operate on sub-tasks? Currently, the logical database change tasks need to perform operations such as termination and skipping on a specific sub-task. Currently, we are achieving this through `modifyJobParameters` by passing sub-task ID and different parameters like `terminate` and `skip`, which is somewhat tricky.

**Comment by LioRoger** [comment]
> The refactor looks good to me. I have another requirement but it's not in emergency: Can we have a mechanism to operate on sub-tasks? Currently, the logical database change tasks need to perform operations such as termination and skipping on a specific sub-task. Currently, we are achieving this through `modifyJobParameters` by passing sub-task ID and different parameters like `terminate` and `skip`, which is somewhat tricky.

that depends on Task design model. currently Task model hopes developer do all job in start method.  no sub task concept is
introduces.  
your requirement is more like a high level concept design.  Imagine if there is a object contains db change tasks, each task operate a instance,  operator delivered by this object to correspond task. but sadly this design is not so urgent.

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**